### PR TITLE
deprecate nagios monitor

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
Deprecate nagios monitor.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,10 @@
 - (Splunk) Add ElasticSearch receiver ([#5165](https://github.com/signalfx/splunk-otel-collector/pull/5165/))
 - (Splunk) Add HAProxy receiver [#5163](https://github.com/signalfx/splunk-otel-collector/pull/5163)
 
+### 🚩Deprecations 🚩
+
+- (Splunk) Deprecate the nagios monitor ([#](https://github.com/signalfx/splunk-otel-collector/pull/))
```
Comment (crobert-1) [inline CHANGELOG.md:12]:
```suggestion
- (Splunk) Deprecate the nagios monitor ([#5172](https://github.com/signalfx/splunk-otel-collector/pull/5172))
```

**Review by jvoravong — APPROVED** [review]

File: internal/signalfx-agent/pkg/monitors/nagios/nagios.go [hunk 2]
```diff
@@ -68,6 +68,8 @@ var (
 // Configure and kick off internal metric collection
 func (m *Monitor) Configure(conf *Config) error {
 	m.logger = logrus.WithFields(logrus.Fields{"monitorType": monitorType, "monitorID": conf.MonitorID})
+	m.logger.Warn("[NOTICE] The nagios monitor is deprecated and will be removed in a future release.")
```
Comment (dmitryax) [inline internal/signalfx-agent/pkg/monitors/nagios/nagios.go:71]: I believe we use `[WARNING]` in other places. Maybe better to be consistent if I'm not missing other places where we do `[NOTICE] `

**Review by dmitryax — APPROVED** [review]

File: internal/signalfx-agent/pkg/monitors/nagios/nagios.go (hunk 2 above)
Comment (atoulme) [reply internal/signalfx-agent/pkg/monitors/nagios/nagios.go:71]: it's already logging at WARN level and therefore will show as a warning. We use [NOTICE] for all deprecation notices so far. 

https://github.com/search?q=repo%3Asignalfx%2Fsplunk-otel-collector%20%5BNOTICE%5D&type=code

You are correct that we use [WARNING], but only in the concept of config converters where we use log.Println:

https://github.com/search?q=repo%3Asignalfx%2Fsplunk-otel-collector+%5BWARNING%5D&type=code

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+### 🚩Deprecations 🚩
+
+- (Splunk) Deprecate the nagios monitor ([#](https://github.com/signalfx/splunk-otel-collector/pull/))
```
Comment (atoulme) [inline CHANGELOG.md:7]:
```suggestion
- (Splunk) Deprecate the nagios monitor ([#5172](https://github.com/signalfx/splunk-otel-collector/pull/5172))
```

**Review by pjanotti — APPROVED** [review]
:shipit:

**Comment by aurbiztondo-splunk** [comment]
@pjanotti - checking with @aunshc about next steps for docs/UI.

**Comment by pjanotti** [comment]
@aurbiztondo-splunk not sure if you recall but here is another deprecation that we did https://github.com/splunk/private-o11y-docs/pull/1699 - I am not sure if in the present case we go with a precise date as in the link.

**Comment by aurbiztondo-splunk** [comment]
Thanks @pjanotti - will check with Aunsh as well! Docs updated in https://github.com/splunk/private-o11y-docs/pull/2292

**Pull request merged**